### PR TITLE
Implementation of PSP3 functionalities

### PR DESCRIPTION
**Pull request by georgitodorov**
This PR includes implementation of the following tasks:

- [GPII-2928: Modify nobody keyed in screen](https://issues.gpii.net/browse/GPII-2928)
- [GPII-2929: Add About screen](https://issues.gpii.net/browse/GPII-2929)
- [GPII-2931: Add sound notifications in PSP](https://issues.gpii.net/browse/GPII-2931)
- [GPII-2932: Display image associated with preference set](https://issues.gpii.net/browse/GPII-2932)
- [GPII-2939: Modify PSP system tray icon tooltips](https://issues.gpii.net/browse/GPII-2939)
- [GPII-2940: Grouping functionality in PSP](https://issues.gpii.net/browse/GPII-2940)
- [GPII-2941: Modify PSP application restart behavior](https://issues.gpii.net/browse/GPII-2941)
- [GPII-2988: Implement click-outside user preference](https://issues.gpii.net/browse/GPII-2988) (PSP-side only. When the Core's end of this functionality is ready, it will be pretty straightforward to "merge" them)
- [GPII-2970: Open PSP panel from icon](https://issues.gpii.net/browse/GPII-2970)

### CODE REVIEW

**Comment by javihernandez** [comment]
ok to test

**Comment by georgitodorov** [comment]
[GPII-2970: Open PSP panel from icon](https://issues.gpii.net/browse/GPII-2970) is now also included in the PR.

File: src/main/gpiiConnector.js [hunk 1]
```diff
@@ -218,25 +276,245 @@ fluid.defaults("gpii.app.dev.gpiiConnector", {
...
+        fluid.each(preferences.settingGroups, function (settingGroup) {
+            applyLivenessFlag(settingGroup.settings);
+        });
     }
 };
+
+/**
+ * An object containing information about the type of the setting and its possible
+ * values, as well as some additional meta information.
+ * @typedef {Object} SettingSchema
```
Comment (amb26) [inline src/main/gpiiConnector.js]: Great use of JSDocs, thanks

**Comment by amb26** [comment]
Thanks, this implementation look like good work to me. A few administrative things - 
i) Could you apply linting since there are a few hundred errors since the last update
ii) Could you try updating to the latest published windows build version 0.3.0-dev.20180507T122135Z.92c7c83 since there have been some changes in core due to merge of Kaspar's lifecycle improvements and we should try to shake down anything that has been disturbed.
I see you have a test failure but it appears to be one in core, so updating may cause the failure to go away/be replaced by a different failure : P

File: testData/grouping/groupingTemplate.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+[
+    {
+        "name": "Windows Hight Contrast",
```
Comment (amb26) [inline testData/grouping/groupingTemplate.json]: Typo "High"

Comment (amb26) [reply testData/grouping/groupingTemplate.json]: Typo remains

Comment (georgitodorov) [reply testData/grouping/groupingTemplate.json]: Ah, thank you! We corrected this typo along with the linting warnings in the ``PSP3-coverage`` branch which also includes Tony's test code coverage work.

**Comment by amb26** [comment]
ok to test

**Comment by amb26** [comment]
Thanks @georgitodorov  and @the-t-in-rtf - it's great to have some accurate coverage reports from this branch/work finally :)
We should have a more thorough pass over this in some separate work but I wanted to just deal with a couple of the biggest holes/easiest wins that the report highlights:

surveyTriggerManager.js - it appears that this check is unnecessary
app.js line 339 - this entire uncaughtExceptionHandler and its calls to the dialogManager is untested
 (there are several other holes in app.js but we can deal with these later)
errorDialog.js line 70 - this IPC event seems never to fire
psp.js line 376 - mysteriously none of these IPC messages "onPSPClose", "onKeyOut", "onSettingAltered" etc. appear to be received - why is this?
settingsBroker.js line 135 - this branch is never executed
utils.js line 77 - this function appears to be unused
   line 215 and line 227 - these IPC utilities appear to never run

In separate work we will have to think how to test some of the PSP's external functionality - e.g. listeners to tray icon, exit handlers, error handlers for websockets messages. Note that also there are no tests for the "update one setting" webSockets message which might require some whitebox testing.

**Comment by georgitodorov** [comment]
I think we can now close this PR as all the code in it is now part of the [QSS v1.1 PR](https://github.com/GPII/gpii-app/pull/62).

**Comment by amb26** [comment]
Closed in favour of https://github.com/GPII/gpii-app/pull/62

**Pull request closed without merging**